Client/mods/deathmatch/logic/CRemoteCalls.cpp

```diff
@@ -28,34 +29,62 @@ CRemoteCalls::~CRemoteCalls()
     m_calls.clear();
 }
 
-void CRemoteCalls::Call(const char* szServerHost, const char* szResourceName, const char* szFunctionName, CLuaArguments* arguments, CLuaMain* luaMain,
+CRemoteCall* CRemoteCalls::Call(const char* szServerHost, const char* szResourceName, const char* szFunctionName, CLuaArguments* arguments, CLuaMain* luaMain,
                         const CLuaFunctionRef& iFunction, const SString& strQueueName, uint uiConnectionAttempts, uint uiConnectTimeoutMs)
 {
-    m_calls.push_back(
-        new CRemoteCall(szServerHost, szResourceName, szFunctionName, arguments, luaMain, iFunction, strQueueName, uiConnectionAttempts, uiConnectTimeoutMs));
-    m_calls.back()->MakeCall();
+    CRemoteCall* pRemoteCall = new CRemoteCall(szServerHost, szResourceName, szFunctionName, arguments, luaMain, iFunction, strQueueName, uiConnectionAttempts, uiConnectTimeoutMs);
+    
+    if (pRemoteCall)
+    {
+        m_calls.push_back(pRemoteCall);
+        m_calls.back()->MakeCall();
+        return pRemoteCall;
+    }
+    return nullptr;
 }
 
-void CRemoteCalls::Call(const char* szURL, CLuaArguments* arguments, CLuaMain* luaMain, const CLuaFunctionRef& iFunction, const SString& strQueueName,
-                        uint uiConnectionAttempts, uint uiConnectTimeoutMs)
+CRemoteCall* CRemoteCalls::Call(const char* szURL, CLuaArguments* arguments, CLuaMain* luaMain, const CLuaFunctionRef& iFunction, const SString& strQueueName,
+                                uint uiConnectionAttempts, uint uiConnectTimeoutMs)
 {
-    m_calls.push_back(new CRemoteCall(szURL, arguments, luaMain, iFunction, strQueueName, uiConnectionAttempts, uiConnectTimeoutMs));
-    m_calls.back()->MakeCall();
+    CRemoteCall* pRemoteCall = new CRemoteCall(szURL, arguments, luaMain, iFunction, strQueueName, uiConnectionAttempts, uiConnectTimeoutMs);
+
+    if (pRemoteCall)
```

Comment:
I don't think `new` will return nullptr

Reply:
Note: `new` throws `std::bad_alloc`, but you can select the non-throwing allocation function by using `std::nothrow`